libr/bin/format/elf/elf.c

```diff
@@ -4047,28 +4055,24 @@ static bool _read_symbols_from_phdr(ELFOBJ *eo, ReadPhdrSymbolState *state) {
 		psym->ordinal = i;
 		psym->in_shdr = false;
 		psym->name[ELF_STRING_LENGTH - 2] = '\0';
-#define PERMIT_UNNAMED_SYMBOLS 0
-#if PERMIT_UNNAMED_SYMBOLS
 		if (!*psym->name) {
+#if PERMIT_UNNAMED_SYMBOLS
 			strcpy (psym->name, "unksym");
-		}
 #else
-		if (!*psym->name) {
 			R_LOG_DEBUG ("empty symbol name", psym->name);
-			ret->len--;
+			// XXX R2_590 ret->len--;
+			RVecRBinElfSymbol_pop_back (ret);
```

Comment:
R_VEC_END_ITER (ret)--;

Reply:
solved it by picking the name before alocating the new entry, so i can just move forward without shrinking the vector unnecessarily